### ISSUE TITLE
public upload overwrites the share owner file content

### ISSUE THREAD

**Issue by kiranparajuli589** [owncloud/ocis#2258]
## Describe the bug
When public overwrites a file in a public link share, sharer file content is overwritten

## Steps to reproduce

1. create a user `uu1`
2. create a folder `simple` using user `uu1`
3. upload a file `lorem.txt` with content `initial content` inside `simple` folder using user `uu1`
4. create a public link share with `read, update, create, delete` permission
5. access the public link as public
6. upload `lorem.txt` file with content `new content` overwriting the file
7. as user `uu1` download the content of file `simple/lorem.txt`

## Expected behavior
- content of `simple/lorem.txt` should not be changed.
- content should be `initial content`

## Actual behavior
- content of `simple/lorem.txt` is overwritten
- content is `new content`

## Setup

Please describe how you started the server and provide a list of relevant environment variables.

<details>
<p>

using either of storage driver

```console
OCIS_VERSION=git
BRANCH=latest
STORAGE_FRONTEND_UPLOAD_DISABLE_TUS=false
```

</p>
</details>

## Additional context
Test is covered [here](https://github.com/owncloud/web/blob/master/tests/acceptance/features/webUIUpload/upload.feature#L151)

**Comment by phil-davis** [comment]
I don't understand what is wrong. The public link share has "full" access. So the public is allowed to overwrite files in the share.

**Comment by kiranparajuli589** [comment]
as per the test linked, with the oc10 backend the owner file content is not overwritten. but with `ocis`, it is overwritten

**Comment by phil-davis** [comment]
`And as "Alice" the content of "simple-folder/lorem.txt" should be the same as the local "lorem.txt"`

That step compares the online content with the content of the local file that was just uploaded.

Maybe we need to re-word that step text so it can be understood more easily?

**Comment by kiranparajuli589** [comment]
> `And as "Alice" the content of "simple-folder/lorem.txt" should be the same as the local "lorem.txt"`
> 
> That step compares the online content with the content of the local file that was just uploaded.
> 
> Maybe we need to re-word that step text so it can be understood more easily?

yes @phil-davis ji, I can do that

**Comment by wkloucek** [comment]
@kiranparajuli589 @phil-davis this looks fixed as owncloud/web#5459 is merged , please reopen if still relevant
